Adopted from 4.3.0 patch to remove downloads during test.

Ultimately the to be downloaded matrices should be in DISTFILES.

Index: hipSPARSE-rocm-5.0.2/clients/tests/CMakeLists.txt
===================================================================
--- hipSPARSE-rocm-5.0.2.orig/clients/tests/CMakeLists.txt
+++ hipSPARSE-rocm-5.0.2/clients/tests/CMakeLists.txt
@@ -23,117 +23,6 @@
 
 find_package(GTest REQUIRED)
 
-# Download some test matrices
-set(TEST_MATRICES
-  SNAP/amazon0312
-  Muite/Chebyshev4
-  FEMLAB/sme3Dc
-  Williams/webbase-1M
-  Bova/rma10
-  JGD_BIBD/bibd_22_8
-  Williams/mac_econ_fwd500
-  Williams/mc2depi
-  Hamm/scircuit
-  Sandia/ASIC_320k
-  GHS_psdef/bmwcra_1
-  HB/nos1
-  HB/nos2
-  HB/nos3
-  HB/nos4
-  HB/nos5
-  HB/nos6
-  HB/nos7
-  DNVS/shipsec1
-)
-
-set(TEST_MD5HASH
-  f567e5f5029d052e3004bc69bb3f13f5
-  e39879103dafab21f4cf942e0fe42a85
-  a95eee14d980a9cfbbaf5df4a3c64713
-  2d4c239daad6f12d66a1e6a2af44cbdb
-  a899a0c48b9a58d081c52ffd88a84955
-  455d5b699ea10232bbab5bc002219ae6
-  f1b0e56fbb75d1d6862874e3d7d33060
-  8c8633eada6455c1784269b213c85ea6
-  3e62f7ea83914f7e20019aefb2a5176f
-  fcfaf8a25c8f49b8d29f138f3c65c08f
-  8a3cf5448a4fe73dcbdb5a16b326715f
-  b203f7605cb1f20f83280061068f7ec7
-  b0f812ffcc9469f0bf9be701205522c4
-  f185514062a0eeabe86d2909275fe1dc
-  04b781415202db404733ca0c159acbef
-  c98e35f1cfd1ee8177f37bdae155a6e7
-  c39375226aa5c495293003a5f637598f
-  9a6481268847e6cf0d70671f2ff1ddcd
-  73372e7d6a0848f8b19d64a924fab73e
-)
-
-if(NOT TARGET hipsparse)
-  set(CONVERT_SOURCE ${CMAKE_SOURCE_DIR}/../deps/convert.cpp)
-else()
-  set(CONVERT_SOURCE ${CMAKE_SOURCE_DIR}/deps/convert.cpp)
-endif()
-
-if(BUILD_ADDRESS_SANITIZER)
-  execute_process(COMMAND ${CMAKE_CXX_COMPILER} ${CONVERT_SOURCE} -O3 -fsanitize=address -shared-libasan -o ${PROJECT_BINARY_DIR}/mtx2csr.exe)
-else()
-  execute_process(COMMAND ${CMAKE_CXX_COMPILER} ${CONVERT_SOURCE} -O3 -o ${PROJECT_BINARY_DIR}/mtx2csr.exe)
-endif()
-
-list(LENGTH TEST_MATRICES len)
-math(EXPR len1 "${len} - 1")
-
-foreach(i RANGE 0 ${len1})
-  list(GET TEST_MATRICES ${i} m)
-  list(GET TEST_MD5HASH ${i} md5)
-
-  string(REPLACE "/" ";" sep_m ${m})
-  list(GET sep_m 0 dir)
-  list(GET sep_m 1 mat)
-
-  # Download test matrices if not already downloaded
-  set(CMAKE_MATRICES_DIR ${PROJECT_BINARY_DIR}/matrices)
-  if(NOT EXISTS "${CMAKE_MATRICES_DIR}/${mat}.bin")
-    message("-- Downloading and extracting test matrix ${m}.tar.gz")
-    file(DOWNLOAD https://sparse.tamu.edu/MM/${m}.tar.gz ${CMAKE_MATRICES_DIR}/${mat}.tar.gz
-        INACTIVITY_TIMEOUT 3
-        STATUS DL)
-
-    list(GET DL 0 stat)
-    list(GET DL 1 msg)
-
-    if(NOT stat EQUAL 0)
-      message("-- Timeout has been reached, trying mirror ...")
-      # Try again using ufl links
-      file(DOWNLOAD https://www.cise.ufl.edu/research/sparse/MM/${m}.tar.gz ${CMAKE_MATRICES_DIR}/${mat}.tar.gz
-          INACTIVITY_TIMEOUT 3
-          STATUS DL)
-
-      list(GET DL 0 stat)
-      list(GET DL 1 msg)
-
-      if(NOT stat EQUAL 0)
-        message(FATAL_ERROR "${msg}")
-      endif()
-    endif()
-
-    # Check MD5 hash before continuing
-    file(MD5 ${CMAKE_MATRICES_DIR}/${mat}.tar.gz hash)
-
-    # Compare hash
-    if(NOT hash STREQUAL md5)
-      message(FATAL_ERROR "${mat}.tar.gz is corrupted")
-    endif()
-    execute_process(COMMAND tar xf ${mat}.tar.gz WORKING_DIRECTORY ${CMAKE_MATRICES_DIR})
-    
-    file(RENAME ${CMAKE_MATRICES_DIR}/${mat}/${mat}.mtx ${CMAKE_MATRICES_DIR}/${mat}.mtx)
-    execute_process(COMMAND ${PROJECT_BINARY_DIR}/mtx2csr.exe ${mat}.mtx ${mat}.bin WORKING_DIRECTORY ${CMAKE_MATRICES_DIR})
-    # TODO: add 'COMMAND_ERROR_IS_FATAL ANY' once cmake supported version is 3.19
-    file(REMOVE_RECURSE ${CMAKE_MATRICES_DIR}/${mat}.tar.gz ${CMAKE_MATRICES_DIR}/${mat} ${CMAKE_MATRICES_DIR}/${mat}.mtx)
-	
-  endif()
-endforeach()
-
 set(HIPSPARSE_TEST_SOURCES
   hipsparse_gtest_main.cpp
   test_axpyi.cpp
